src/rstar.jl

```diff
@@ -40,21 +40,22 @@ function rstar(
         throw(ArgumentError("training and test data subsets must not be empty"))
 
     xtable = _astable(x)
+    ycategorical = MLJModelInterface.categorical(ysplit)
```

Comment:
Models that implement `reformat` are required to also implement `selectrows` for the outputs. Since reformatting could be expensive, we can perform the reformatting upfront:
```suggestion
    ycategorical = MLJModelInterface.categorical(ysplit)
    xdata, ydata = MLJModelInterface.reformat(classifier, _astable(x), ycategorical)
```

Reply:
Ah OK, I didn't know that (seems generally the MLJ docs for developers that want to work with MLJModelInterface are a bit lacking, or maybe I just didn't study them carefully enough?).

Reply:
There are a few things missing from the docs, but it's mostly that they're quite dense and jump around a bit, so it's easy to miss a requirement. The relevant bit from https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use/#Implementing-a-data-front-end is
> To specify how the model-specific representation of data is to be resampled, implement selectrows(model::SomeModelType, I, data...) -> resampled_data for each overloading of reformat(model::SomeModel, args...) -> data above. Here I is an arbitrary abstract integer vector or : (type Colon).